integration-tests/app_id_test.go

```diff
@@ -165,17 +151,17 @@ func (suite *AppCredentialTestSuite) TestUnlimitedUserKey() {
 	fmt.Printf("Response: %v", res)
 	assert.Equal(suite.T(), 200, res.StatusCode, "Invalid http response code for user_key unlimited test: %v", res.StatusCode)
 
-	if err := DeleteUserKey(suite.ServiceID, "unlimited_app_id", suite.UserKey); err != nil {
+	if err := suite.backend.Pop(); err != nil {
 		suite.Errorf(err, "Failed to delete Application's user key: %v")
 	}
-	if err := DeleteApplication(suite.ServiceID, "unlimited_app_id"); err != nil {
+	if err := suite.backend.Pop(); err != nil {
 		suite.Errorf(err, "Failed to delete applications: %v")
 	}
 }
 
 func (suite *AppCredentialTestSuite) TestUnlimitedAppId() {
 	// Add a new unlimited app
-	if err := AddApplication(suite.ServiceID, "unlimited_app_id", suite.PlanID); err != nil {
+	if err := suite.backend.Push("app", []interface{}{suite.ServiceID, "unlimited_app_id", suite.PlanID}); err != nil {
 		suite.Errorf(err, "Error adding an application: %v")
```

Comment:
`suite.Errorf` is just a wrapper around the error. Use `require.Nilf` or `assert.Nilf` to fail the test in case of an error. `require` stop and fails the test right after the error and `assert` fails the test after executing the rest of the test. In this case `assert` suites more but if one of the deletions failed, then we will have issues with the successor test cases because the apisonator already contains apps, services etc from the previous test cases. To overcome this issue we can randomly generate the app_ids, service_ids from the code itself. for now just adding a assert or require seems okay.

Reply:
Sure, I'll change this and yes, I am thinking of adding a new PR allowing changes mentioned by you.

Reply:
either will work. only difference is assert will delete the rest of the resources.